shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1630,6 +1631,26 @@ - (void)setupKeyboardAnimationVsyncClient {
   [_keyboardAnimationVSyncClient await];
 }
 
+- (void)keyboardAnimationUpdateViewportMetricsInFrameRasterizedCallback {
+  fml::closure platformTask = [weakSelf = [self getWeakPtr]] {
+    if (!weakSelf) {
+      return;
+    }
+    fml::scoped_nsobject<FlutterViewController> flutterViewController(
+        [(FlutterViewController*)weakSelf.get() retain]);
+    if (!flutterViewController) {
+      return;
+    }
+    [flutterViewController updateViewportMetrics];
+  };
+
+  flutter::Shell& shell = [_engine.get() shell];
+  fml::closure onFrameRasterizedCallback = [&shell, platformTask] {
+    shell.GetTaskRunners().GetPlatformTaskRunner()->PostTask(platformTask);
```

Comment:
I believe this also needs to be synchronized. The rasterizer needs to wait until the platform task to finish before starting next frame.